packages/vue-components/src/panels/PanelBase.js

```diff
@@ -141,6 +141,9 @@ export default {
           Thus, we need to reset the maxHeight to its current height for collapse transition to work.
         */
         this.$refs.panel.style.maxHeight = `${this.$refs.panel.scrollHeight}px`;
+        if (this.$refs.panel.scrollHeight > window.innerHeight) {
```

Comment:
> @ang-zeyu I am not sure if the disorienting issue can be resolved even by removing the transition effect. Trying it out locally with the changes from this branch:

I don't think this implementation is correct, @tlylt the case in your gif shouldn't occur ⬇️ 

> if top of panel beyond viewport (I think there's some existing checks for this too) 

the relevant existing check is here https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/panels/PanelBase.js#L147

Reply:
> For pages like the one below, with a panel that is not too long, I prefer the previous closing animation since it looks fine to me. With the new changes, the closing is more abrupt and it may be a little surprising if another shorter panel nearby has an animation and suddenly the next panel doesn't have it

If corrected, does it account for the 2 links you linked too?

If not, we could try https://github.com/MarkBind/markbind/pull/1965#discussion_r922768335, or even remove the closing transitions completely for UX consistency (⭐I think this might be better based on what you've mentioned⭐).

Again this is all very subjective, and I'm not huge fan of large animations (fwiw I'm the original implementor of this feature, haven't found the closing transitions anything apart from dizzying)
Our primary [issue](https://github.com/MarkBind/markbind/issues/486) back then was also that the page did not scroll back up to the top of the panel *at all*, much less the transitions which was a cool "afterthought" =P

If agreed here perhaps we should just remove closing transitions completely (including the jquery scroll).
@damithc @MarkBind/active-devs can chime in too.

maybe give a thumbs up 👍 or 👎 on this comment to get a consensus. :-)

Reply:
I'm OK to remove the transition.

Reply:
@eeliana based on the 👍 so far, I think we can go ahead with replacing any and all closing transitions (including the jquery scroll) with an "instant snap" back to the top of the panel. The opening ones can be kept.

Reply:
@ang-zeyu 
> If corrected, does it account for the 2 links you linked too?

Yes if all closing transitions are removed, then I think the behavior is uniform and I am ok with that.

> I don't think this implementation is correct, @tlylt the case in your gif shouldn't occur ⬇️

Re looking at this again, may need your advice on if there's anything special going on here... So the case in my gif is one that the closed panels are near the bottom of the page. When opening or closing, the height of the page could reduce or increase such that the panel moves away from the original position. Base on my observation, whether before or after the latest change done by @eeliana, it seems like at times the panel could move away from the original position and result in slight disorienting effect, or somehow it won't. Here's a few recordings:
1. You can see that the first closing of "Guideline: Maximum readability" result in the panel heading moving, while the open and closing of the subsequent times didn't cause that (which is what we want to achieve?). This is done with the change in this PR applied
![2OJTbHhksz](https://user-images.githubusercontent.com/41845017/181918108-201e477e-9aad-4f30-8d28-71da2a641c4f.gif)

2. You can also experiment with this on https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html which is the original behavior:
a) works fine (seems like I am able to trigger this, by closing the panels one by one, avoiding any scrolling before I close all the panels)
![3KeCqFsO4U](https://user-images.githubusercontent.com/41845017/181918293-4b87e552-17d3-4f13-9f0b-6909416293e0.gif)
b) doesn't really work well (seems like I am able to trigger this, by scrolling around, before closing the panels)
![sGEKNWxtzh](https://user-images.githubusercontent.com/41845017/181918332-282840ec-2e14-4190-8491-75ffec2d340d.gif)

@eeliana Please feel free to experiment a little and follow up on this since it's your PR :)

Reply:
> Re looking at this again, may need your advice on if there's anything special going on here...

I actually thought the issue you wanted to show in your gif was that the closing transition was not activating previously, in that case it should since the top of the panel is in view. (the check isn't correct for that)

> Yes if all closing transitions are removed, then I think the behavior is uniform and I am ok with that.

But not so much relevant since we're going with this now. 😶 

> Re looking at this again, may need your advice on if there's anything special going on here...

This seems separate, let's put up a another issue for this.

Since its occuring only on the first open, likely:
- there is some dynamically loaded content (e.g. `<panel src="...">`) inside that panel
- something to do with how we handle these updates (see [here](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/panels/PanelBase.js#L188) and [here](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/panels/NestedPanel.vue#L98)) and/or how chrome handles these viewport changes (on that note, maybe we could try different browsers too)